### PR TITLE
getting data from API

### PR DESCRIPTION
**Pull request by fernanda1411**

### CODE REVIEW

File: client/src/components/App.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* {
+
+}
```
Comment (van-hsieh) [inline client/src/components/App.scss]: Could you please add a new line at the end of the file?

File: client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.scss [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  .answer-wrapper {
+    display: flex;
+    h3 {
+      margin-right: 10px;
+    }
+    .answer-list {
+      width: 100%;
+    }
+  }
+}
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.scss]: Could you please add a new line at the end of the file? Just a tiny best practice thing.

File: client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.jsx [hunk 3]
```diff
@@ -3,14 +3,56 @@ import './QuestionListItem.scss';
 import QuestionListItemAnswer from '../QuestionListItemAnswer/QuestionListItemAnswer.jsx';
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.jsx:3]: I see your scss file but it is not imported into the React component. Is it intentional? If yes, ignore this comment :)

File: client/src/components/QuestionsAndAnswers/QuestionListItemAnswer/QuestionListItemAnswer.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.answer {
+  margin-bottom: 10px;
+  border-bottom: 1px solid gray;
+  padding-bottom: 10px;
+  p.answer-text {
+   margin-bottom: 10px;
+   line-height: 1.5;
+ }
+}
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionListItemAnswer/QuestionListItemAnswer.scss]: Could you please add a new line at the end of the file?

File: client/src/components/QuestionsAndAnswers/QuestionListItemAnswerActions/QuestionListItemAnswerActions.scss [hunk 5]
```diff
@@ -1,3 +1,16 @@
...
+    padding: 0;
+    text-decoration: underline;
+    background-color: transparent;
+    font-size: 12px;
+    color: gray;
+  }
+  .separator {
+    margin: 0 10px;
+  }
 }
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionListItemAnswerActions/QuestionListItemAnswerActions.scss]: Could you please add a new line at the end of the file?

File: client/src/components/QuestionsAndAnswers/QuestionListItemAnswerImages/QuestionListItemAnswerImages.scss [hunk 6]
```diff
@@ -1,3 +1,7 @@
 .answer-images {
   display: flex;
+  .answer-image img{
+    width: 100px;
+    margin: 10px;
+  }
 }
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionListItemAnswerImages/QuestionListItemAnswerImages.scss]: Could you please add a new line at the end of the file?

File: client/src/components/QuestionsAndAnswers/QuestionListItemQuestion/QuestionListItemQuestion.scss [hunk 7]
```diff
@@ -0,0 +1,9 @@
+.question {
+  display: flex;
+  width: 100%;
+  .question-text {
+    flex-grow: 1;
+    font-weight: bold;
+    font-size: 18px;
+  }
+}
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionListItemQuestion/QuestionListItemQuestion.scss]: Could you please add a new line at the end of the file?

File: client/src/components/QuestionsAndAnswers/QuestionListItemQuestionActions/QuestionListItemQuestionActions.scss [hunk 8]
```diff
@@ -1,3 +1,17 @@
...
+  }
+  .add-answer-button{
+    border: none;
+    padding: 0;
+    text-decoration: underline;
+    background-color: transparent;
+    font-size: 12px;
+    color: gray;
+  }
 }
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionListItemQuestionActions/QuestionListItemQuestionActions.scss]: Could you please add a new line at the end of the file?

File: client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.scss [hunk 9]
```diff
@@ -1,4 +1,15 @@
...
+    list-style: none;
+  }
+
   border: 1px solid black;
   padding: 10px;
+
+  h2 {
+    width: 100%;
+  }
 }
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.scss]: Could you please add a new line at the end of the file?

File: client/src/components/QuestionsAndAnswers/QuestionsList/QuestionsList.jsx [hunk 10]
```diff
@@ -2,12 +2,32 @@ import React from 'react';
...
     <>
-      <ul>
-        <QuestionListItem />
-        <QuestionListItem />
+      <ul className="question-list">
+        {questionsAndAnswers.map(qa => {
+          console.log('Question List QA', qa);
+          return (
+            <QuestionListItem
+              key={qa.question_id}
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionsList/QuestionsList.jsx:18]: The code might be easier to read if you just past the entire qa as an object into the sub-component. So you don't have to keep track of all the variables.

File: client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.jsx [hunk 11]
```diff
@@ -3,14 +3,56 @@ import './QuestionListItem.scss';
...
+      </div>
+      { !!Object.keys(answers).length && <div className="answer-wrapper">
+        <h3>A:</h3>
+        <ul className="answer-list">
+          {Object.keys(answers).map(answerId => {
+            const answer = answers[answerId];
+
+            return (
+              <QuestionListItemAnswer
+                key={answer.id}
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.jsx:43]: Passing in the entire answer object might reduce the number of lines in the code. Just a suggestion for minor improvement.

File: client/src/components/QuestionsAndAnswers/SearchQuestions/SearchQuestions.scss [hunk 12]
```diff
@@ -0,0 +1,7 @@
+.search-question {
+  margin: 20px 0 30px;
+  input {
+    width: 100%;
+    padding: 10px;
+  }
+}
```
Comment (van-hsieh) [inline client/src/components/QuestionsAndAnswers/SearchQuestions/SearchQuestions.scss]: Please add a new line here

File: client/src/components/Shared/HelpfulButton/HelpfulButton.jsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <div className="helpful-button">
+      <p>Helpful?</p>
+      <button onClick={handleClick}>Yes</button>
+      <p>({helpfulness})</p>
+    </div>
+  );
+};
+
+export default HelpfulButton;
```
Comment (van-hsieh) [inline client/src/components/Shared/HelpfulButton/HelpfulButton.jsx]: Please add a new line here

File: client/src/components/Shared/HelpfulButton/HelpfulButton.scss [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  button {
+    border: none;
+    padding: 0;
+    text-decoration: underline;
+    background-color: transparent;
+    margin: 0 5px;
+    font-size: 12px;
+    color: gray;
+  }
+}
```
Comment (van-hsieh) [inline client/src/components/Shared/HelpfulButton/HelpfulButton.scss]: Please add a new line here

**Review by van-hsieh — CHANGES_REQUESTED** [review]

File: client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.jsx (hunk 3 above)
Comment (fernanda1411) [reply client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.jsx:3]: The scss files are imported to the components. I see that we are doing it on slightly different ways but the way I'm doing also works.

File: client/src/components/QuestionsAndAnswers/QuestionsList/QuestionsList.jsx (hunk 10 above)
Comment (fernanda1411) [reply client/src/components/QuestionsAndAnswers/QuestionsList/QuestionsList.jsx:18]: I see your point and I agree with it but I prefer to exposed all the props to the components consumer and this way we can be sure that all the required data is passed in.

File: client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.jsx (hunk 11 above)
Comment (fernanda1411) [reply client/src/components/QuestionsAndAnswers/QuestionListItem/QuestionListItem.jsx:43]: same as above

**Pull request merged**